### PR TITLE
refactor(mkdir): Move to clap + add a test

### PR DESCRIPTION
**Pull request by sylvestre**

### CODE REVIEW

File: src/uu/mkdir/src/mkdir.rs [hunk 1]
```diff
@@ -120,7 +131,7 @@ fn mkdir(path: &Path, recursive: bool, mode: u16, verbose: bool) -> i32 {
     }
 
     if verbose {
-        show_info!("created directory '{}'", path.display());
+        println!("{}: created directory '{}'", executable!(), path.display());
```
Comment (rivy) [inline src/uu/mkdir/src/mkdir.rs:134]: Is this change needed? Doesn't `show_info!(...)` do the same thing and in a project standardized way?

**Review by rivy — CHANGES_REQUESTED** [review]
Nits with question about `println!()` substitution for `show_info!()`.

File: src/uu/mkdir/src/mkdir.rs (hunk 1 above)
Comment (sylvestre) [reply src/uu/mkdir/src/mkdir.rs:134]: because show_info sends message to stderr. 
println! send to stdout like gnu mkdir

**Comment by sylvestre** [comment]
As it was only nit and the CI is green, I will land it now.
We can always replace println! later.
